lib/presentation/auth/change_password/confirm/bloc/change_password_confirm_cubit.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_template/data/auth/model/auth_data_error.dart';
+import 'package:flutter_template/data/auth/repository/auth_repository.dart';
+import 'package:flutter_template/data/auth/service/remote/model/forgot_password_confirm/forgot_password_confirm_request_model.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_error.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_state.dart';
+import 'package:flutter_template/util/extensions/string_extension.dart';
+import 'package:flutter_template/util/mixins/resettable_bloc_mixin.dart';
+
+class ChangePasswordConfirmCubit extends Cubit<ChangePasswordConfirmState>
+    with ResettableBlocMixin {
+  final AuthRepository _authRepository;
+
+  ChangePasswordConfirmCubit(this._authRepository)
+      : super(
+          const ChangePasswordConfirmStateInitial(
+            email: "",
+            password: "",
+            confirmPassword: "",
+            code: "",
+          ),
+        );
+
+  @override
+  void resetState() {
+    emit(
+      ChangePasswordConfirmStateInitial(
+        email: state.email,
+        password: state.password,
+        confirmPassword: state.confirmPassword,
+        code: state.code,
+      ),
+    );
+  }
+
+  @override
+  void onChange(Change<ChangePasswordConfirmState> change) {
+    super.onChange(change);
+
+    // TODO: Add Analytics and logging here
+  }
```

Comment:
For logging prefer using a global `BlocObserver` or logs in individual methods, depending on the chosen strategy.

Analytics are very case-by-case basis, so that might not be the ideal place.

Reply:
Also careful about logging globally the values inside the state, as it may contain sensitive data or very long data